iOS Client/DevotionsController/DevotionsCoordinator.swift

```diff
@@ -25,13 +26,53 @@ class DevotionsCoordinator: NSObject, Coordinator, StoreSubscriber {
         let devotionsController = MBDevotionsViewController.instantiateFromStoryboard()
         navigationController.pushViewController(devotionsController, animated: true)
         MBStore.sharedStore.subscribe(self)
+        let store = MBStore()
+        let devotions = store.getDevotions()
+        if devotions.count == 0 {
+            store.syncDevotions { syncedDevotions, error in
+                if error != nil {
+                    MBStore.sharedStore.dispatch(LoadedDevotions(devotions: Loaded.error))
```

Comment:
To be safe, these two dispatches should be in a dispatch.main.async block to guarantee that they occur on the main thread, since that is a requirement of ReSwift. In this case, I think it would be on the main thread b/c our client method to load the devotions is probably synchronous, but if we changed that out to be a network call then we might get some random undefined behavior stemming from these dispatches occurring on a background thread.

Reply:
good call I should put that first dispatch in an async block